### PR TITLE
HOCS-6251: Create document extract

### PR DESCRIPTION
**Pull request by JosephRogerson**

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @BeforeEach
+    public void setup() {
+        zonedDateTimeConverter = new ZonedDateTimeConverter();
+
+        given(infoClient.getCaseTypes()).willReturn(Set.of(new CaseTypeDto("Test", "a1", "TEST")));
+    }
+
+    @Test
+    public void shouldReturnExport() throws IOException {
+        documentExportService.export(LocalDate.of(2020, 1, 1), LocalDate.now().plusDays(1), outputStream, "TEST",
```
Comment (jeff-horton-ho-sas) [inline src/test/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportServiceTest.java:33]: There are no `DOCUMENT_CREATED` events in [setup.sql](https://github.com/UKHomeOffice/hocs-audit/blob/b4b3b15d83fdfb518a90580a7e67c6b08580a32f/src/test/resources/export/setup.sql) so this isn't fully testing that events can be mapped to an export row. Note: it looks like `getCsvDataRows(result)` doesn't exclude the header row, hence that returning size = 1.

**Review by jeff-horton-ho-sas — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportService.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    @Override
+    protected ExportDataConverter getDataConverter(boolean convert, CaseTypeDto caseType) {
+        if (!convert) {
+            return new ExportDataConverter();
+        }
+
+        Map<String, String> uuidToName = new HashMap<>();
+
+        uuidToName.putAll(
```
[the remark this answers is not in the document]
Comment (JosephRogerson) [reply src/main/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportService.java]: I've removed those mappings, I believe the info and casework client are still needed as dependencies as they are used in the super constructor

File: src/test/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportServiceTest.java (hunk 1 above)
Comment (jeff-horton-ho-sas) [reply src/test/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportServiceTest.java:36]: I'd still like to see some actual data being mapped in the test. Consider adding something like 

```
(5, '00000000-0000-0000-0000-000000000005', '10000000-0000-0000-0000-000000000000',
 '20000000-0000-0000-0000-000000000000', '30000000-0000-0000-0000-000000000004', 'service-1',
 '{"documentUUID": "30000000-0000-0000-0000-000000000000", "documentType": "documentType", "documentTitle": "Document title"}',
 'env-1', '2020-01-01 01:00:00.000000', 'DOCUMENT_CREATED', '40000000-0000-0000-0000-000000000000', 'a1');
```

to [setup.sql](https://github.com/UKHomeOffice/hocs-audit/blob/b4b3b15d83fdfb518a90580a7e67c6b08580a32f/src/test/resources/export/setup.sql) and check that the 2nd data row includes the expected values?

Comment (JosephRogerson) [reply src/test/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportServiceTest.java:36]: I think I may be missing something, is that not what I'm already checking here on line 47?

Comment (jeff-horton-ho-sas) [reply src/test/java/uk/gov/digital/ho/hocs/audit/service/DocumentExportServiceTest.java:36]: Apologies that change wasn't showing for me when I reviewed it. LGTM 👍

**Pull request merged**